modules/dynamic-apiserver/pkg/app/app.go

```diff
@@ -116,3 +121,110 @@ func (srv *Service) SubscriptionAlreadyExist(ctx context.Context, subscriptionId
 	}
 	return len(services.Items) > 0, nil
 }
+
+func (srv *Service) CreateKuberlogicServiceBackup(ctx context.Context, serviceId *string) (*kuberlogiccomv1alpha1.KuberlogicServiceBackup, error) {
+	klb := &kuberlogiccomv1alpha1.KuberlogicServiceBackup{
+		ObjectMeta: v1.ObjectMeta{
+			Name: fmt.Sprintf("%s-%d", *serviceId, time.Now().Unix()),
+			Labels: map[string]string{
+				util.BackupRestoreServiceField: *serviceId,
+			},
+		},
+		Spec: kuberlogiccomv1alpha1.KuberlogicServiceBackupSpec{
+			KuberlogicServiceName: *serviceId,
+		},
+	}
+	err := srv.kuberlogicClient.Post().
+		Resource(backupK8sResource).
+		Name(klb.GetName()).
+		Body(klb).
+		Do(ctx).
+		Into(klb)
+	return klb, err
+}
+
+func (srv *Service) DeleteKuberlogicServiceBackup(ctx context.Context, backupId *string) error {
+	return srv.kuberlogicClient.Delete().
+		Resource(backupK8sResource).
+		Name(*backupId).
+		Do(ctx).
+		Error()
+}
+
+func (srv *Service) WaitForServiceBackup(ctx context.Context, serviceId *string, backupId *string, maxRetries int) error {
+	timeout := time.Second
+	for i := maxRetries; i > 0; i-- {
+		backups, err := srv.ListKuberlogicServiceBackupsByService(ctx, serviceId)
+		if err != nil {
+			return err
+		}
+		for _, backup := range backups.Items {
+			if backup.GetName() == *backupId {
+				switch backup.Status.Phase {
+				case kuberlogiccomv1alpha1.KlbSuccessfulCondType:
+					return nil
+				case kuberlogiccomv1alpha1.KlbFailedCondType:
+					return errors.New("Error occured while creating service backup")
+				}
+			}
+		}
+		timeout = timeout * 2
+		time.Sleep(timeout)
+		srv.log.Infof("backup is not ready, trying after %s. Left %d retries", timeout, i-1)
+	}
+	return errors.New("Retries exceeded, backup is not ready")
+}
+
+/*
+	Archive service will:
+	1. Take new backup of a service (if backups enabled)
+	2. Remove all previous backups
+	3. Remove the service itself
+*/
+func (srv *Service) ArchiveKuberlogicService(serviceId *string) {
```

Comment:
This is not covered by tests, right?

Reply:
From current apiserver perspective: I believe it will be relatively hard thing to do, since our current test-suite uses single request-response per mocked client and `ArchiveKuberlogicService` uses more than one
I didn't find any possibility to make it from operator perspective either

But, anyway, I may missed something and open to suggestions

Reply:
Agree, we need more complex mocking function for different requests under the hood. Ok, leave it for now, I'll create task to add it.